Add inside rust blog post on imposter syndrome (#958)
rust-lang/blog.rust-lang.org: posts/inside-rust/2022-04-13-imposter-syndrome.md

* [Jane Lusby](https://github.com/yaahc/): "I frequently struggle with imposter
  syndrome and feeling like I don't get as much done as my peers. When I do all
  of my work based off of notifications I completely lose track of what I've
  done and end up starving the tasks I wanted to work on. I'm learning to set
  reasonable expectations for myself, getting better at managing distractions,
  and being intentional about when I respond to github/zulip notifications
  which helps me with keeping track of what I've done and making steady
  progress on my priorities."
* [Josh Triplett](https://github.com/joshtriplett/): "I didn't fully understand
  `Pin` until I read fasterthanlime's ["Pin and
  suffering"](https://fasterthanli.me/articles/pin-and-suffering) blog post and
  I gave a talk in 2016 where my [most important
  point](https://www.youtube.com/watch?v=U8Gl3RTXf88#t=24m40s) was that people
  erroneously believe that you have to be an expert to write an RFC or change
  Rust, but that I wasn't, and you don't need to be one either."
* [Ralf Jung](https://github.com/ralfjung): "I am still surprised anyone is
  taking Miri and Stacked Borrows seriously."

> 

Reviewer: Here's a section from me that I hope fits alright :)

```suggestion
* [Forest Anderson](https://github.com/angelonfira/): "As someone who just
  learned last week what `dyn` does, it still amazes me that I have something to
  give as a team lead. I was immersed in Rust communities by writing weekly
  blogs about Veloren (I took this on because I didn't know enough to contribute
  code), which lead to helping with the Rust Gamedev newsletter, which led me to
  helping to run the Cross Team Collaboration Fun Times meetup!"

```

project has layer[^4] after layer[^5] of safeguards to make sure we have a
chance to catch and fix them before they affect our users. These incidents are
unavoidable, expected, and honestly fine! This is the most fundamental
philosophy of both the Rust language and the Rust project: we don't think it's
sufficient to build robust systems by only including people who don't make
mistakes; we think it's better to provide tooling and process to catch and
prevent mistakes. It isn't an accident that our motto is "A language empowering
everyone to build reliable and efficient software." We want people to feel
empowered to make changes they're not 100% confident in, to make mistakes, to
learn, and to grow within the Rust project. This is how all of us got to where
we are today!

So, if you look up to people within the Rust project, if the work we do here
interests you, if you have always wanted to contribute, and _especially_ if you
already have contributed, I want you to know that you're one of the people we
want to apply for [Rust Foundation grants and
fellowships](https://foundation.rust-lang.org/grants/). You're one of the

> people we want to eventually see join teams. You're not an imposter, and I want

Reviewer 2: > You're one of the people we want to *eventually see join* teams.

(emphasis mine) 

I'd be careful with this. I'm not sure that there aren't people this post can address and help who aren't already on a team. Imposter syndrome can even impact "insiders". Let's not forget them as well!

---
layout: post
title: "Imposter Syndrome"
author: Jane Lusby Project Director of Collaboration
team: Rust Foundation Project Directors <https://foundation.rust-lang.org/about/>
---

*Preface: This is in response to some feedback the project directors received
from the Rust Foundation staff. Some of the contributors they'd talked to said
they didn't feel justified in applying for Foundation grants even though they'd
love the opportunity, because they don't feel qualified or deserving of them
compared to the other amazing contributors they look up to within the Rust
project. This was a little bit heart breaking to me, because I know exactly
what that feeling is like[^1], and also know just how wrong they are.*

Imposter syndrome is an insidious problem within software communities. Many of
us, especially members of marginalized communities, struggle to shake the
feeling that we aren't as qualified as our peers. This makes us feel
unqualified and undeserving compared to those around us. It can make us
hesitate to join communities in the first place. For those already involved, it
can create a sense of impending doom where you constantly feel like you're

> going to get found out and expelled from the community. It overall just not

Reviewer 3: Think there's a minor typo here, perhaps:
```suggestion
going to get found out and expelled from the community. It's overall just not
```
or
```suggestion
going to get found out and expelled from the community. It overall is just not
```
or

```suggestion
going to get found out and expelled from the community. Overall it's just not
```

> great for mental health, 0/10 would not recommend.

Reviewer 3: This confused me at first as I read as a double negative of sorts wherein zero out of ten wouldn't recommend (since that could mean ten would?).

Would a comment help clarify or have I simply misunderstood?

```suggestion
great for mental health, 0/10, would not recommend.
```

---
layout: post
title: "Imposter Syndrome"

> author: Jane Lusby Project Director of Collaboration

Reviewer 4: ```suggestion
author: Jane Lusby, Project Director of Collaboration
```

unqualified and undeserving compared to those around us. It can make us
hesitate to join communities in the first place and, for those already
involved, it can create a sense of impending doom where you constantly feel
like you're going to get found out and expelled from the community. Overall
it's just not great for mental health, 0/10, would not recommend.

The thing is though, imposter syndrome is a logical fallacy[^2]. Imposter
syndrome occurs when we discount what we know and inflate what we think other
people know, and this effect is often then reinforced by systemic bias for
those of us who don't get the assumption of competence.

![picture of imposter syndrome, left side shows a large circle saying "What I think others know" and a small circle inside of it saying "What I know", right side shows the same small circle saying "What I know" surrounded by many other equally sized small circles labeled "What others know"](/static/images/2022-04-13-imposter-syndrome/imposter_syndrome.jpg)

In reality, we're all specialists within the Rust project. We all have areas
where we have deep expertise and other large areas where we only have the

> vaguest idea of how things work. Niko still comes to me to ask questions about

Reviewer 5: it may be worth annotating Niko's name with a link to his Github

Reviewer 5: or stating his title(s)

error handling. I frequently need to tell my fellow contributors that I have no
idea what the acronyms they're using mean[^3]. But this doesn't mean we don't
deserve our positions within the project. We don't expect every contributor
know everything, to be perfect, or to make no mistakes. The only thing we
expect from our contributors is the ability to collaborate effectively with
others and the ability to learn and grow over time.

The thing that makes the Rust project as good as it is isn't a couple of
prolific contributors lifting mountains by themselves, it's everyone working
together that brought us to where we are today. We all make mistakes. The
project has layer[^4] after layer[^5] of safeguards to make sure we have a

> chance to catch and fix them before they affect our users. These incidents are

Reviewer 6: Also, even if all of these fail, we have procedures to handle such cases and fix such mistakes . Const promotion, I'm looking at you!

unavoidable, expected, and honestly fine! We want people to feel empowered to
make changes they're not 100% confident in, to make mistakes, to learn, and to
grow within the Rust project. This is how all of us got to where we are today!

So, if you look up to people within the Rust project, if the work we do here
interests you, if you have always wanted to contribute, and _especially_ if you
already have contributed, I want you to know that you're one of the people we
want to apply for [Rust Foundation grants and
fellowships](https://foundation.rust-lang.org/grants/). You're one of the
people we want to eventually see join teams. You're not an imposter, and I want
you to know that I really look forward to seeing you around the project.

---

> *To help reinforce and normalize this, I want to turn this end of this blog

Reviewer 5: I like this idea but I wonder if there's a better spot to put this? I think reviewing might get overwhelming and eventually folks will not look at this blogpost. I think that maybe a microsite for the grants program could be a great location for this content!

Reviewer 6: Yes, that's a great idea. I have so many cool anecdotes that belong here!

> post into a living documenent of times when current of past project members

Reviewer 6: ```suggestion
post into a living document of times when current or past project members
```
